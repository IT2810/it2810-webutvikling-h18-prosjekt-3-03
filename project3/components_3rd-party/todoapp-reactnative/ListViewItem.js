import React, {Component} from 'react';
import {TouchableHighlight, View, Text} from 'react-native';
import {Button, Icon} from 'native-base';
import CheckBox from './CheckBox';

class ListViewItem extends Component {
    constructor(props) {
        super(props);
        this._onCheckBoxPressed = this._onCheckBoxPressed.bind(this);
        this.state = {
            data: this.props.data
        }
    }

    componentWillReceiveProps(props) {
        this.setState({
            data: props.data
        })
    }

    _onCheckBoxPressed() {
        const data = this.state.data;
        data.completed = !data.completed;
        this.setState({
            data: data
        });

        this.props.onCompletedChange(data, this.props.dataIndex);
    }

    render() {
        let data = this.state.data;
        let color = data.completed ? '#C5C8C9' : '#000';
        let textDecorationLine = data.completed ? 'line-through' : 'none';
        return (
            <TouchableHighlight underlayColor={'#eee'} style={{
                paddingTop: 6,
                paddingBottom: 6,
                backgroundColor: "#F8F8F8",
                borderBottomWidth: 1,
                borderColor: '#eee'
            }} {...this.props.sortHandlers}>
                <View style={{flexDirection: 'row', alignItems: 'center'}}>
                    <CheckBox data={data} color={color} onCheckBoxPressed={this._onCheckBoxPressed}/>
                    <Text
                        style={{fontSize: 18, color: color, textDecorationLine: textDecorationLine}}>{data.title}</Text>
                    <Button
                        transparent
                        onPress={() => this.props.onRemove(this.props.dataIndex)}
                        style={{marginLeft: 'auto'}}>
                        <Icon name={"trash"}/>
                    </Button>
                </View>
            </TouchableHighlight>
        )
    }
}

module.exports = ListViewItem;
