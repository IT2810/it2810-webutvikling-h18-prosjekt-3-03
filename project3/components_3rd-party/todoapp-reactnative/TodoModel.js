class TodoModel {
    constructor(title, completed, createdAt = new Date()) {
        this.title = title;
        this.completed = completed || false;
        this.createdAt = createdAt;
    }
}

module.exports = TodoModel;
