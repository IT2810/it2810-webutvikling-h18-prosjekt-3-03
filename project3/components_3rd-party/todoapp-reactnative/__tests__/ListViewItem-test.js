import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';
import ListViewItem from '../ListViewItem';
import TodoModel from '../TodoModel';

it('renders correctly', () => {
    // TODO: test shallowly to avoid having to recreate snapshot if subcomponents change
    const tree = renderer
        .create(<View>
            <ListViewItem data={new TodoModel("Test title", false, new Date(0))}
                          dataIndex={0}/>
            <ListViewItem data={new TodoModel("Test title 2", true, new Date(0))}
                          dataIndex={0}/>
        </View>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
