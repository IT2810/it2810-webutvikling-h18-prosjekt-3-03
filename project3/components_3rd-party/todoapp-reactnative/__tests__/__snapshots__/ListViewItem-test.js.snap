// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomWidth": 1,
        "borderColor": "#eee",
        "paddingBottom": 6,
        "paddingTop": 6,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          data={
            TodoModel {
              "completed": false,
              "createdAt": 1970-01-01T00:00:00.000Z,
              "title": "Test title",
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "padding": 8,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "borderRadius": 5,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "textDecorationLine": "none",
          }
        }
      >
        Test title
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": "auto",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomWidth": 1,
        "borderColor": "#eee",
        "paddingBottom": 6,
        "paddingTop": 6,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          data={
            TodoModel {
              "completed": true,
              "createdAt": 1970-01-01T00:00:00.000Z,
              "title": "Test title 2",
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "padding": 8,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "borderRadius": 5,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#C5C8C9",
            "fontSize": 18,
            "textDecorationLine": "line-through",
          }
        }
      >
        Test title 2
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": "auto",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;
